### PR TITLE
feat(graphql): vpc support for sql lambda

### PR DESCRIPTION
**Pull request by sundersc**
#### Description of changes

Deploy SQL Lambda in VPC if the database is in VPC. 

##### CDK / CloudFormation Parameters Changed

NA

#### Issue #, if available

NA

#### Description of how you validated changes
- Manual test
- Updated E2E test

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Any CDK or CloudFormation parameter changes are called out explicitly

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-graphql-model-transformer/src/resolvers/rds/resolver.ts [hunk 1]
```diff
@@ -136,11 +154,18 @@ export const createRdsLambda = (
     ],
     lambdaRole,
     environment,
-    undefined,
+    Duration.seconds(30),
     stack,
+    sqlLambdaVpcConfig,
```
Comment (sundersc) [inline packages/amplify-graphql-model-transformer/src/resolvers/rds/resolver.ts:159]: This is the intended change on this file. Most of the other changes are lint fixes.

File: packages/amplify-graphql-model-transformer/src/resolvers/rds/resolver.ts [hunk 2]
```diff
@@ -166,40 +191,60 @@ export const createRdsLambdaRole = (roleName: string, stack: Construct, secretEn
...
+
+  role.addToPolicy(
+    new PolicyStatement({
+      effect: Effect.ALLOW,
+      resources: ['*'],
+      actions: [
+        'ec2:CreateNetworkInterface',
+        'ec2:DescribeNetworkInterfaces',
+        'ec2:DeleteNetworkInterface',
+      ],
```
Comment (sundersc) [inline packages/amplify-graphql-model-transformer/src/resolvers/rds/resolver.ts:211]: 'ec2:DescribeNetworkInterfaces' permission is required on resources '*' for lambda deployment in VPC.
https://docs.aws.amazon.com/lambda/latest/dg/configuration-vpc.html#vpc-permissions

File: packages/amplify-category-api/src/provider-utils/awscloudformation/utils/rds-resources/database-resources.ts [hunk 3]
```diff
@@ -188,3 +187,26 @@ export const removeVpcSchemaInspectorLambda = async (context: $TSContext): Promi
...
+
+  const databaseConfig: ImportedDataSourceConfig = await databaseConfigurationInputWalkthrough(engine);
+  return {
+    secrets: {
+      engine,
+      ...databaseConfig,
+    },
+    storeSecrets: true,
+  };
+};
```
Comment (sundersc) [inline packages/amplify-category-api/src/provider-utils/awscloudformation/utils/rds-resources/database-resources.ts:212]: This function is moved from generate-schema.ts to here.

File: packages/amplify-category-api/src/graphql-transformer/transform-graphql-schema-v2.ts [hunk 4]
```diff
@@ -258,6 +274,21 @@ const _buildProject = async (context: $TSContext, opts: TransformerProjectOption
   }
 };
 
+const isSqlLambdaVpcConfigRequired = async (
```
Comment (alharris-at) [inline packages/amplify-category-api/src/graphql-transformer/transform-graphql-schema-v2.ts:277]: N.p. but this method name suggests I'll get a boolean back, not a 'something or nothing'. Maybe rename to `getSqlDataSourceLambdaVpcConfig` ?

**Review by alharris-at — APPROVED** [review]
Change looks good to me. Only a nitpick on method naming.

**Review by phani-srikar — APPROVED** [review]

**Pull request merged**